tests/unit/test_saml.py

```diff
@@ -70,20 +70,132 @@ def test_saml_with_invalid_url(_):
 
 
 @patch("urllib.request.urlopen")
-@pytest.mark.parametrize(
-    "metadata_file,binding",
-    [
-        ("metadata_1.xml", "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"),
-        ("metadata_2.xml", "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Post"),
-    ],
-)
-def test_saml_with_valid_metadata(urlopen_mock, metadata_file, binding):
+def test_saml_with_valid_signed_metadata(urlopen_mock):
+    """
+    arrange: mock the metadata contents so that they valid.
+    act: access the metadata properties.
+    assert: the properties are populated as defined in the metadata.
+    """
+    with open("tests/unit/files/metadata_signed.xml", "rb") as metadata:
```

Comment:
Would https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text work for this?